src/index.tsx

```diff
@@ -26,9 +28,27 @@ const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement)
 
 root.render(
   <React.StrictMode>
-    <QueryClientProvider client={queryClient}>
-      <RouterProvider router={router} />
-      <ReactQueryDevtools initialIsOpen={false} />
-    </QueryClientProvider>
+    <HelmetProvider>
+      <QueryClientProvider client={queryClient}>
+        {(!!customAppTitle || !!customAppDescription) && (
+          <Helmet>
```

Comment:
The idea of handling all these things conditionally looks error prone to me. Add HelmetProvider, but don't use it always, add Helmet conditionally, conditional meta tags, values from env, i18n or HTML.

We don't need to support dynamic meta tags. Can't we set meta tags during build time in index.html?

Reply:
>  Can't we set meta tags during build time in index.html?

Would be nice, and less error prone, that's for sure.

Reply:
OK, I got rid of Helmet, and using the `posthtml-expression` parcel plugin to write those meta tags at compile time.